default/scripting/ship_parts/Shield/SH_ROBOTIC_INTERFACE_SHIELDS.focs.txt

```diff
@@ -25,7 +25,7 @@ Part
             activation = ([[ROBOTIC_SHIELD_EFFECT]] >= [[BEST_SHIELD_EFFECT]])
             stackinggroup = "ROBOTIC_SHIELD_PART_STACK"
             accountinglabel = "ROBOTIC_INTERFACE_EFFECT"
-            effects = SetMaxShield value = Value - [[BEST_SHIELD_EFFECT]] + [[ROBOTIC_SHIELD_EFFECT]]
+            effects = SetMaxShield value = Value - [[BEST_SHIELD_EFFECT]] + ( [[ROBOTIC_SHIELD_EFFECT]] * (GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR") )
```

Comment:
Why `GameRule name = "RULE_SHIP_WEAPON_DAMAGE_FACTOR"` not `[[SHIP_WEAPON_DAMAGE_FACTOR]]` or `[[SHIP_SHIELD_FACTOR]]` ?

Reply:
Did not know what style to prefer and if macro expansion would work everywhere (was before asking the parens question in the forum). Later on it became clear that using a macro name is more appropriate as I was reusing the same rules for different purposes (e.g. damage rule for shield scaling) so decoupling intent and the used rule became necessary. Will change all those rules use to macros for consistency.

Reply:
Refactor to macros done